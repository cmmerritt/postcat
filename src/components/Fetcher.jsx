/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './fetcher.css';

const Fetcher = ({ urlText, method, reqText, onChange, onSubmit }) => {
  return <div className={styles.fetcher}>
    <form onSubmit={onSubmit}>
      {/* <label htmlFor="urlText"></label> */}
      <input data-testid="urlText" className={styles.searchInput} id="urlText" placeholder="URL" type="text" name="urlText" value={urlText} onChange={onChange}/>
      <div className={styles.radioButtons}>
        <input type="radio" id="get" name="method" value="GET" method={method} onChange={onChange}/>
        <label htmlFor="get">GET</label>

        <input type="radio" id="post" name="method" value="POST" method={method} onChange={onChange}/>
        <label htmlFor="post">POST</label>

        <input type="radio" id="put" name="method" value="PUT" method={method} onChange={onChange}/>
        <label htmlFor="put">PUT</label>

        <input type="radio" id="patch" name="method" value="PATCH" method={method} onChange={onChange}/>
        <label htmlFor="patch">PATCH</label>

        <input type="radio" id="delete" name="method" value="DELETE" method={method} onChange={onChange}/>
        <label htmlFor="delete">DELETE</label>
      </div>

      <div>
        <button type="submit" className={styles.goButton}>Go!</button>
      </div>
    </form>

    <form>
      {/* <label htmlFor="reqText"></label> */}
      <textarea data-testid="reqText" id="reqText" placeholder="Raw JSON Body" type="text" name="reqText" value={reqText} onChange={onChange}/>
    </form>
  </div>;
};

Fetcher.propTypes = {
  urlText: PropTypes.string.isRequired,
  method:  PropTypes.string.isRequired,
  reqText: PropTypes.any,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func
};

export default Fetcher;
