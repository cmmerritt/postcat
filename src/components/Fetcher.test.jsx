import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Fetcher from './Fetcher.jsx';

describe('Fetcher component', () => {
  afterEach(() => cleanup());
  it('renders Fetcher', () => {
    const { asFragment } = render(<Fetcher />);
    expect(asFragment()).toMatchSnapshot();
  });
});
