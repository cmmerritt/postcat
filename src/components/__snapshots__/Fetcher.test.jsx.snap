// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetcher component renders Fetcher 1`] = `
<DocumentFragment>
  <div
    class="fetcher"
  >
    <form>
      <input
        class="searchInput"
        data-testid="url"
        id="url"
        name="url"
        placeholder="URL"
        type="text"
        value=""
      />
      <div
        class="radioButtons"
      >
        <input
          id="get"
          name="method"
          type="radio"
          value="GET"
        />
        <label
          for="get"
        >
          GET
        </label>
        <input
          id="post"
          name="method"
          type="radio"
          value="POST"
        />
        <label
          for="post"
        >
          POST
        </label>
        <input
          id="put"
          name="method"
          type="radio"
          value="PUT"
        />
        <label
          for="put"
        >
          PUT
        </label>
        <input
          id="patch"
          name="method"
          type="radio"
          value="PATCH"
        />
        <label
          for="patch"
        >
          PATCH
        </label>
        <input
          id="delete"
          name="method"
          type="radio"
          value="DELETE"
        />
        <label
          for="delete"
        >
          DELETE
        </label>
      </div>
      <div>
        <button
          class="goButton"
          type="submit"
        >
          Go!
        </button>
      </div>
    </form>
    <form>
      <textarea
        data-testid="body"
        id="body"
        name="body"
        placeholder="Raw JSON Body"
        type="text"
      />
    </form>
  </div>
</DocumentFragment>
`;
